### ISSUE TITLE
Cannot connect to generated URL in Quick Start

### ISSUE THREAD

**Issue by DavidScales** [novnc/noVNC#1215]
**Describe the bug**
I cannot connect to the URL generated by the [README "Quick Start" instructions](https://github.com/novnc/noVNC#quick-start)

**Reproducing & context**

I'm using an AWS ec2 Ubuntu 18.04 instance, with a GUI (`ubuntu-desktop gnome-panel gnome-settings-daemon metacity nautilus gnome-terminal xfce4`) and a vnc server (`vnc4server`) installed. 

I `ssh` into the machine, & start the vnc server with `vncserver -geometry 1340x750`, which I confirm is working, by connecting via a desktop client (tigerVNC) with `<ec2_elastic_ip>:5901` & my vnc password.

I clone the noVNC repo into the machine, and when I run `./utils/launch.sh --vnc localhost:5901`, I get the following output:

```
Warning: could not find self.pem
Using local websockify at /home/ubuntu/Downloads/noVNC/utils/websockify/run
Starting webserver and WebSockets proxy on port 6080
/home/ubuntu/Downloads/noVNC/utils/websockify/websockify/websocket.py:30: UserWarning: no 'numpy' module, HyBi protocol will be slower
  warnings.warn("no 'numpy' module, HyBi protocol will be slower")
WebSocket server settings:
  - Listen on :6080
  - Web server. Web root: /home/ubuntu/Downloads/noVNC
  - No SSL/TLS support (no cert file)
  - proxying from :6080 to localhost:5901

Navigate to this URL:

    http://ip-172-31-2-201:6080/vnc.html?host=ip-172-31-2-201&port=6080

Press Ctrl-C to exit
```

Which looks successful to me. The only warnings present seem dismissible
* I don’t have a `self.pem` file, but the ["Quick Start" in the README](https://github.com/novnc/noVNC#quick-start) doesn’t say anything about needing that
* I don’t have a cert file, but I don’t need SSL right now
* I apparently don’t have numpy but I’m okay with that

However, when I visit the URL in my local browser, it simply doesn’t connect. No errors logged or anything, just the same result that you’d expect if you typed in `some-non-exisitant-url.com` into your browser:
![screen shot 2019-03-06 at 5 59 16 pm](https://user-images.githubusercontent.com/16726377/53927344-68fa6800-403b-11e9-8833-38b8da61ab24.png)

I have all ports open for incoming and outgoing traffic on my ec2 instance (to ensure it’s not a firewall issue).

Any ideas? Seems like a simple use case so I can’t imagine I’m not simply missing something. I've tried checking though the bugs but didn’t find an issue this broad. I'm relatively inexperienced with Linux and this sort of thing.

**Expected behavior**
I expect the URL in my local browser to connect & show the VM desktop

**Client:**
 - OS: Local machine is mac (Mojave 10.14.3), remote machine is AWS ec2 Ubuntu 18.04
 - Browser: Trying on Chrome 72 and Firefox 65

**Server:**
 - noVNC version: just pulled a couple days ago, commit ID 9a823732a01af080357669d12d355cdc8e7e099c
 - VNC server: vnc4server 
 - WebSocket proxy: websockify provided by `./utils/launch.sh` download

**Comment by samhed** [comment]
This isn't a noVNC or websockify issue.

Even if you open up your local firewall on your ubuntu machine, the amazon firewalls could still be blocking you. You need to make sure that you are using the public hostname provided for your EC2 server. For me I remember that it looked something like this `ec2-01-234-567-890.us-west-2.compute.amazon.com`

**Comment by pm-pm** [comment]
Use the public IP Address. (As used with VNC Client) Can be seen on AWS Console.

If you don´t use a static IP (extra Costs may occur), the Public IP changes after each shutdown. (No change on reboot). Advice: Use DynDNS Service for access.
AWS uses a NAT, so the local tools on the server can often not determine the correct external/public IP

If you are "inside" your AWS Server Grid you can use the internal IP address. The same problem should occur at home.

**Comment by samhed** [comment]
> Maybe a Note in the help information could make it more clear to beginners.

You mean adding information on how to access AWS machines in noVNC's wiki/README?

**Comment by pm-pm** [comment]
I mean accessing machines behind a NAT (like AWS) or  with port forwarding (like home).
In this case the noVNC Setup script only shows the network internal ip. This is formally correct, but can not be used to connect from outside. From outside the network you can not connect with this hostname. You need to use the external IP.

This is typically NOT a noVNC issue, but a note can help the user to find his mistake faster.

**Comment by DavidScales** [comment]
Awesome! That totally worked :D 

For posterity, I was able to connect with the exact same procedure, simply substituting the noVNC outputted URL:

`http://ip-172-31-2-201:6080/vnc.html?host=ip-172-31-2-201&port=6080`

with `http://MY_PUBLIC_IP:6080/vnc.html?host=MY_PUBLIC_IP&port=6080`

where `MY_PUBLIC_IP` is listed in the AWS console in the format described by @samhed, `ec2-01-234-567-890.us-west-2.compute.amazon.com`. 

It also works with `MY_PUBLIC_IP` as the static IP of my instance, to which @pm-pm alludes. This is called an "elastic IP" in AWS terminology. 

I think a note of some kind would be helpful, if not in the README, perhaps in an FAQ or similar. I can't imagine I am the only person to try noVNC with a product like AWS, and I had actually tried noVNC a couple months back and gave up on it when I couldn't figure this out. 

Feel free to close. Thanks so much for your help, I'm excited to use this tool!
